pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/HLRealtimeSegmentDataManager.java

```diff
@@ -377,10 +377,9 @@ public void run() {
 
           try {
             _segmentLogger.info("Marking current segment as completed in Helix");
-            RealtimeSegmentZKMetadata metadataToOverwrite = new RealtimeSegmentZKMetadata();
+            SegmentZKMetadata metadataToOverwrite = new SegmentZKMetadata(segmentZKMetadata.getSegmentName());
             metadataToOverwrite.setTableName(_tableNameWithType);
-            metadataToOverwrite.setSegmentName(realtimeSegmentZKMetadata.getSegmentName());
-            metadataToOverwrite.setSegmentType(SegmentType.OFFLINE);
```

Comment:
hmm.. is this a typo in the old code, or intentionally?

Reply:
Seems a typo to me. It does not matter actually because segment type is already deprecated and is only used to determine the key for download url, which does not apply to the HLC segment.

Reply:
I think we used to claim only consuming segment as real-time segment and all persisted(completed) segments are in `OFFLINE` type.
If we don't have any logic rely on this, then it should be fine. E.g. segment retention.

Reply:
I'll keep it the same as before in case it is used (I don't see any usage though).